File: model/src/main/scala/ru/mipt/acsl/decode/model/domain/impl/naming/FqnImpl.scala
package ru.mipt.acsl.decode.model.domain.impl.naming

import ru.mipt.acsl.decode.model.domain.pure.{naming => n}

/**
  * @author Artem Shein
  */
private case class FqnImpl(parts: Seq[n.ElementName]) extends n.Fqn {
  def copyDropLast: n.Fqn = FqnImpl(parts.dropRight(1))
}

File: model/src/main/java/ru/mipt/acsl/decode/model/naming/FqnImpl.java
package ru.mipt.acsl.decode.model.naming;

import java.util.List;

/**
 * Created by metadeus on 06.06.16.
 */
public class FqnImpl implements Fqn {

    private final List<ElementName> parts;

    @Override
    public List<ElementName> getParts() {
        return parts;
    }

    public Fqn copyDropLast() {
        return new FqnImpl(parts.subList(0, parts.size() - 1));
    }

    FqnImpl(List<ElementName> parts) {
        this.parts = parts;
    }

}
